### PR TITLE
backend: Add support for randomized choices

### PR DESCRIPTION
**Pull request by matevz**
Adds support for deterministic randomized question choices based on the contract private key + coupon code + question index.

Removes correctChoice field from the question config since the first choice in the choice list is now always the correct one.

### CODE REVIEW

File: backend/contracts/Quiz.sol [hunk 1]
```diff
@@ -40,8 +44,8 @@ contract Quiz {
     bytes32 _key;
     // List of questions.
     QuizQuestion[] _questions;
-    // List of correct answer choices.
-    uint8[] _questionsCorrectChoices;
+    // Total number of choices. Use for generating the permutation vector.
```
Comment (aefhm) [inline backend/contracts/Quiz.sol]:
```suggestion
    // Total number of choices. Used for generating the permutation vector.
```

File: backend/contracts/Quiz.sol [hunk 2]
```diff
@@ -128,9 +139,67 @@ contract Quiz {
...
+        uint k=0; // Number of processed choices in total.
+        for (uint i=0; i<_questions.length; i++) {
+            bytes32 seed = keccak256(abi.encode(_key, coupon, i));
+            // Enumerate choices.
+            for (uint8 j=0; j<_questions[i].choices.length; j++) {
+                pv[k+j] = j;
+            }
+            // Permute choices.
+            for (uint j=0; j<_questions[i].choices.length; j++) {
+                uint8 tmp = pv[k+uint8(seed[j%seed.length])%_questions[i].choices.length];
```
Comment (aefhm) [inline backend/contracts/Quiz.sol]: `seed.length` is fixed?

Comment (aefhm) [inline backend/contracts/Quiz.sol]: Maybe we could name `k+uint8(seed[j%seed.length])%_questions[i].choices.length`. I needed a sec to grok this (potentially just a me problem though).

**Review by aefhm — APPROVED** [review]

File: backend/contracts/Quiz.sol (hunk 2 above)
Comment (matevz) [reply backend/contracts/Quiz.sol]: Right. Currently it's `bytes32`, so we can permute 32 choices. The 33rd one will use the 0th permutation again.

Comment (aefhm) [reply backend/contracts/Quiz.sol]: K. No problems. Just wondering about declaring constants, but probably not worth it.

**Pull request merged**